src/librustdoc/html/static/storage.js

```diff
@@ -27,14 +27,15 @@ function removeClass(elem, className) {
 function onEach(arr, func, reversed) {
     if (arr && arr.length > 0 && func) {
         var length = arr.length;
+        var i;
```

Comment:
Is this really necessary? Each `i` is local to the loop so declaring it out here makes reasoning the loop harder.

Reply:
JS `var`s don't seem to understand the principle of scope. :)

Reply:
In JS a `var` is hoisted to the top of the function. The new `let` and `const` don't have this behavior.

Reply:
re: fold, yes, browsers implemented Array.prototype.reduce() a long time ago.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce

Much like `Iter` and `for` loops are equivalent in Rust, `reduce` and `map` are equivalent to `for` loops in JS for similar reasons.